### ISSUE TITLE
Should we support configuring the OSes go-test runs on?

### ISSUE THREAD

**Issue by galargh** [protocol/.github#349]
So far, I've encountered one instance where this would be a useful feature - see https://github.com/libp2p/go-openssl/pull/25/files#r909792651

**Comment by marten-seemann** [comment]
Yes, that would be helpful. Is there any way to do it without requiring an extra step? It is nice that the GitHub Actions scheduler knows the matrix in advance right now.

**Comment by galargh** [comment]
Unfortunately, I don't think it is at the moment. I found an issue that would make it possible though and followed up on it - https://github.com/actions/runner/issues/1180

**Comment by galargh** [comment]
OSes can now be skipped through https://github.com/pl-strflt/uci#job-specific-configuration-files.

Now that the go-test workflow is a reusable workflow, we'll also be able to make use of reusable workflow inputs for configuration. With that, we'd be able to skip jobs for skipped OSes entirely instead of going through the runner setup even though we're not doing any work. 

This will be tracked in https://github.com/pl-strflt/uci/issues/42.
